src/Engine/Party.cpp

```diff
@@ -973,16 +973,20 @@ void Party::GivePartyExp(unsigned int pEXPNum) {
                 if (player.conditions.HasNone({CONDITION_UNCONSCIOUS, CONDITION_DEAD, CONDITION_PETRIFIED, CONDITION_ERADICATED})) {
                     pLearningPercent = player.getLearningPercent();
                     playermodexp = pEXPNum + pEXPNum * pLearningPercent / 100;
-                    player.experience += playermodexp;
-                    if (player.experience > 4000000000) {
-                        player.experience = 0;
-                    }
+                    SetCharacterXP(player, player.experience + playermodexp);
                 }
             }
         }
     }
 }
 
+void Party::SetCharacterXP(Character &player, int xp) {
```

Comment:
This should be a method of `Character`?

Reply:
Yes that's just makes more sense. I'll move it there.